operations/src/main/java/edu/unc/lib/boxc/operations/impl/importxml/ImportXMLJob.java

```diff
@@ -394,30 +372,94 @@ private void processUpdates(MultiDestinationTransferSession session, PID resumeP
                                             "Error reading or converting MODS stream: " + ex.getMessage());
                                 } catch (NotFoundException ex) {
                                     failed.put(currentPid.getQualifiedId(), "Object not found");
+                                } catch (OptimisticLockException ex) {
+                                    failed.put(currentPid.getQualifiedId(), ex.getMessage());
                                 } catch (FedoraException ex) {
                                     failed.put(currentPid.getQualifiedId(),
                                             "Error retrieving object from Fedora: " + ex.getMessage());
                                 }
-                            } else if (foundResumptionPoint) {
-                                return;
                             }
                         } else {
-                            if (!resumeMode && updateOperation) {
+                            if (dsOpDetails != null) {
                                 // Store all of the content from the incoming document
-                                xmlWriter.add(e);
+                                dsOpDetails.addContent(e);
                             }
                         }
                         break;
                 }
             }
         } finally {
-            if (xmlWriter != null) {
-                xmlWriter.close();
+            if (dsOpDetails != null) {
+                dsOpDetails.close();
             }
         }
 
     }
 
+    /**
+     * Start a datastream operation, if it passes preconditions:
+     *     * Has a valid type attribute (currently only MODS)
+     *     * Either has no last modified attribute, or has one in ISO 8601 format
+     *     * Is a valid operation (currently only updates)
+     * @param element
+     * @return Details about the operation, or null if it is not a valid operation
+     */
+    private DatastreamOperationDetails startDatastreamOperation(StartElement element) {
+        Attribute operation = element.getAttributeByName(operationAttribute);
+        if (operation == null || !BulkXMLConstants.OPER_UPDATE_ATTR.equals(operation.getValue())) {
+            return null;
+        }
+        DatastreamOperationDetails details = new DatastreamOperationDetails();
```

Comment:
does this need to be created at this point? can it be created further down when it's needed, after the typeAttr checks?